symengine/tests/basic/test_functions.cpp

```diff
@@ -4081,3 +4082,42 @@ TEST_CASE("test_dummy", "[Dummy]")
     CHECK(neq(*xdummy1, *x1));
     CHECK(neq(*xdummy1, *x1->as_dummy()));
 }
+
+TEST_CASE("test_sign", "[Sign]")
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const Basic> r = sign(one);
+    CHECK(eq(*r, *one));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(zero);
+    CHECK(eq(*r, *zero));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(real_double(1.2));
+    CHECK(eq(*r, *one));
+
+    r = sign(real_double(-1.2));
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(Nan);
+    CHECK(eq(*r, *Nan));
+
+    r = sign(complex_double(std::complex<double>(0, 3)));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *I));
+
+    r = sign(complex_double(std::complex<double>(0, -3)));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *mul(I, minus_one)));
```

Comment:
@isuruf Can you please tell me why is this check and the one above return `sign(0.0 + 3.0*I)` and `sign(0.0 - 3.0*I)` even though the cases are being handled in the method?